spring-cloud-gcp-starters/spring-cloud-gcp-starter-logging/src/main/java/org/springframework/cloud/gcp/logging/autoconfig/StackdriverLoggingAutoConfiguration.java

```diff
@@ -38,6 +39,6 @@
 	@Bean
 	@ConditionalOnMissingBean
 	public TraceIdLoggingWebMvcInterceptor getLoggingWebMvcInterceptor() {
-		return new TraceIdLoggingWebMvcInterceptor();
+		return new TraceIdLoggingWebMvcInterceptor(new XCloudTraceIdFromRequestExtractor());
```

Comment:
Since we're using dependency injection, we shouldn't be explicitly creating an instance of ` XCloudTraceIdFromRequestExtractor` like that. Instead, I would modify this method to:

```java
public TraceIdLoggingWebMvcInterceptor getLoggingWebMvcInterceptor(TraceIdFromRequestExtractor extractor)
```
and add:
```java
@Bean
@ConditionalOnMissingBean
public TraceIdFromRequestExtractor traceIdFromRequestExtractor() {
    return new XCloudTraceIdFromRequestExtractor();
}
```
This would enable the users to override just the `TraceIdFromRequestExtractor`, rather than the whole `TraceIdLoggingWebMvcInterceptor`.

Reply:
We may also provide a configuration option `trace.extractor=XCloud/Zipkin/XCloud-Zipkin/XCloud-Zipkin` to allow to reconfigure existing extractors just by the configuration property.